src/utilities/int_array.c

```diff
@@ -138,6 +138,135 @@ hypre_IntArrayCloneDeep( hypre_IntArray *x )
    return hypre_IntArrayCloneDeep_v2(x, hypre_IntArrayMemoryLocation(x));
 }
 
+/*--------------------------------------------------------------------------
+ * hypre_IntArrayMigrate
+ *--------------------------------------------------------------------------*/
+
+HYPRE_Int
+hypre_IntArrayMigrate( hypre_IntArray      *v,
+                       HYPRE_MemoryLocation memory_location )
+{
+   HYPRE_Int            size                = hypre_IntArraySize(v);
+   HYPRE_Int           *v_data              = hypre_IntArrayData(v);
+   HYPRE_MemoryLocation old_memory_location = hypre_IntArrayMemoryLocation(v);
+
+   HYPRE_Int           *w_data;
+
+   /* Update v's memory location */
+   hypre_IntArrayMemoryLocation(v) = memory_location;
+
+   if ( hypre_GetActualMemLocation(memory_location) !=
+        hypre_GetActualMemLocation(old_memory_location) )
+   {
+      w_data = hypre_TAlloc(HYPRE_Int, size, memory_location);
+      hypre_TMemcpy(w_data, v_data, HYPRE_Int, size,
+                    memory_location, old_memory_location);
+      hypre_TFree(v_data, old_memory_location);
+      hypre_IntArrayData(v) = w_data;
+   }
+
+   return hypre_error_flag;
+}
+
+/*--------------------------------------------------------------------------
+ * hypre_IntArrayPrint
+ *--------------------------------------------------------------------------*/
+
+HYPRE_Int
+hypre_IntArrayPrint( MPI_Comm        comm,
+                     hypre_IntArray *array,
+                     const char     *filename )
+{
+   HYPRE_Int             size            = hypre_IntArraySize(array);
+   HYPRE_MemoryLocation  memory_location = hypre_IntArrayMemoryLocation(array);
+
+   hypre_IntArray       *h_array;
+   HYPRE_Int            *data;
+
+   FILE                 *file;
+   HYPRE_Int             i, myid;
+   char                  new_filename[1024];
+
+   hypre_MPI_Comm_rank(comm, &myid);
+
+   /* Move data to host if needed*/
+   h_array = (hypre_GetActualMemLocation(memory_location) == hypre_MEMORY_DEVICE) ?
+             hypre_IntArrayCloneDeep_v2(array, HYPRE_MEMORY_HOST) : array;
+   data = hypre_IntArrayData(h_array);
+
+   /* Open file */
+   hypre_sprintf(new_filename, "%s.%05d", filename, myid);
+   if ((file = fopen(new_filename, "w")) == NULL)
+   {
+      hypre_error_w_msg(HYPRE_ERROR_GENERIC, "Error: can't open output file\n");
+      return hypre_error_flag;
+   }
+
+   /* Print to file */
+   hypre_fprintf(file, "%d\n", size);
+   for (i = 0; i < size; i++)
+   {
+      hypre_fprintf(file, "%d %d\n", i, data[i]);
```

Comment:
@victorapm , is there a particular reason you want to print the index, i, to file? Doesn't seem necessary to me, but maybe you have a good reason for it.

Reply:
Hi Wayne, there was no particular reason and I agree it's unnecessary. I'll remove it. Can you check if this builds fine with SYCL? Thanks!

Reply:
@victorapm , I just checked, and it builds OK with sycl. I see no issues with the implementation for sycl.